core/frontend/src/components/vehiclesetup/PwmSetup.vue

```diff
@@ -384,10 +412,9 @@ export default Vue.extend({
         console.warn('Disarming failed!')
       }, 5000)
     },
-    arm_disarm_switch_change(): void {
-      this.desired_armed_state = !this.is_armed
+    arm_disarm_switch_change(new_state: boolean): void {
```

Comment:
Can we have new_state with a better name that reflects the arm state ?
Like.. is_armed ?

Reply:
this is not what it means. this is the state of the switch itself, which we use to arm/disarm.
maybe `wants_to_be_armed` would be better?

Reply:
`should_arm` ?